projects/igniteui-angular/src/lib/directives/date-time-editor/date-time-editor.directive.ts

```diff
@@ -407,13 +432,21 @@ export class IgxDateTimeEditorDirective extends IgxMaskDirective implements OnCh
     if (!value) { return false; }
```

Comment:
you can extract two more methods here `lessThanMinValue` and `biggerThanMaxValue`.
Use both these new methods to check if value is in range, and also use them `validate` method